bcgov/biohubbc
# SIMSBIOHUB-348: Delete single or many observation records

## Links to Jira Tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-348

## Description of Changes

- Adds a new `ObservationTableContext`
- Adds an endpont and repository methods to delete one or more observation records
- Adds `loading` button props to the `YesNoDialog`
- Refactors the observation context to support deleting one or more observation records

## Testing Notes

You should be able to:
 - Create observation records
 - Delete single observation records (using the trash can icon)
 - Delete single observation records (by selecting the row by checkbox, opening the context menu, and clicking **Delete Observation**
 - Delete multiple observation records (by selecting the rows by checkbox, opening the context menu, and clicking **Delete Observations**
